loom-daemon breaks after worktree cleanup: editable install points to deleted path

## Bug

After a Loom worktree is cleaned up (e.g., `loom-clean` or manual removal), `loom-daemon` fails with:

```
ModuleNotFoundError: No module named 'loom_tools'
```

## Root Cause

When `loom-tools` is `pip install -e`'d from a worktree path (e.g., `.loom/worktrees/issue-2178/loom-tools`), and that worktree is later cleaned up after the PR merges, the editable install's source path is deleted. `pip show loom-tools` still reports it as installed, but the `Editable project location` points to a dead path:

```
$ pip show loom-tools
Location: /opt/homebrew/lib/python3.14/site-packages
Editable project location: /Users/rwalters/GitHub/loom/.loom/worktrees/issue-2178/loom-tools
```

Python can't import the module because the source directory no longer exists.

## Reproduction

1. Install Loom in a target repository
2. Have a builder work on loom-tools from a worktree (editable install happens)
3. Merge the PR and clean up the worktree
4. Run `/loom` - daemon fails with `ModuleNotFoundError`

## Implementation Plan

### 1. Add editable install detection to `clean.py`

**File**: `loom-tools/src/loom_tools/clean.py`

Add a safety check in `clean_worktrees()` that detects if any Python environment has an editable install pointing into the worktree about to be removed.

- Add helper function `find_editable_pip_installs(worktree_path)` that runs `pip show loom-tools` across discoverable Python interpreters and checks if `Editable project location` is inside the worktree path
- Add `skipped_editable: int = 0` to `CleanupStats` dataclass
- Insert the check in `clean_worktrees()` after the in-use marker check (early, cheap — before hitting GitHub API)
- The check runs in all modes (not just safe mode) since a broken pip install is always dangerous
- `--force` bypasses the check (with a warning logged)

Pattern to follow — existing safety checks in `clean_worktrees()`:
```python
# CHECK 1: In-use marker (line 298)
# CHECK 2: Active processes (line 311) — skipped in force mode
# CHECK 3: Issue state via GitHub (line 319)
# NEW CHECK: Editable pip install — after CHECK 2, before CHECK 3
```

### 2. Add verification to `loom-daemon.sh`

**File**: `defaults/scripts/loom-daemon.sh`

When falling through to system-installed `loom-daemon` (line 87-89), verify the Python import works before `exec`ing:

```bash
elif command -v loom-daemon &>/dev/null; then
    # Verify system-installed loom-daemon can actually import loom_tools
    if ! loom-daemon --help &>/dev/null 2>&1; then
        echo "[ERROR] loom-daemon is installed but cannot run." >&2
        echo "  This usually means an editable install source was deleted." >&2
        echo "  See: https://github.com/rjwalters/loom/issues/2192" >&2
        echo "" >&2
        echo "  Fix: pip install -e /path/to/loom/loom-tools" >&2
        exit 1
    fi
    exec loom-daemon ${args[@]+"${args[@]}"}
```

### 3. Add test coverage

**File**: `loom-tools/tests/test_clean.py`

- Test `find_editable_pip_installs()` with mocked subprocess output
- Test that `CleanupStats` includes `skipped_editable` field

## Workaround

Reinstall from the canonical source:

```bash
pip install -e /path/to/loom/loom-tools
```

Or record the source path manually in the target repo:

```bash
echo "/path/to/loom" > .loom/loom-source-path
```

--- a/defaults/scripts/loom-daemon.sh
+++ b/defaults/scripts/loom-daemon.sh
@@ -85,7 +85,19 @@ if [[ -n "$LOOM_TOOLS" ]] && [[ -x "$LOOM_TOOLS/.venv/bin/loom-daemon" ]]; then
     # Use venv from loom-tools directory
     exec "$LOOM_TOOLS/.venv/bin/loom-daemon" ${args[@]+"${args[@]}"}
 elif command -v loom-daemon &>/dev/null; then
-    # System-installed
+    # System-installed - verify loom_tools can be imported
+    if ! python3 -c "import loom_tools" 2>/dev/null; then
+        echo "[ERROR] loom-daemon is installed but cannot import loom_tools." >&2
+        echo "" >&2
+        echo "  This usually means the editable install source directory was deleted" >&2
+        echo "  (e.g., a worktree was removed while loom-tools was pip install -e'd from it)." >&2
+        echo "" >&2
+        echo "  To fix:" >&2
+        echo "    1. Reinstall loom-tools: pip install loom-tools" >&2
+        echo "    2. Or reinstall from source: pip install -e /path/to/loom/loom-tools" >&2
+        echo "" >&2
+        exit 1
+    fi
     exec loom-daemon ${args[@]+"${args[@]}"}
 else
     echo "[ERROR] Python daemon not available." >&2

--- a/loom-tools/src/loom_tools/clean.py
+++ b/loom-tools/src/loom_tools/clean.py
@@ -13,6 +13,7 @@ Exit codes:
 from __future__ import annotations
 
 import argparse
+import json
 import pathlib
 import re
 import shutil
@@ -43,6 +44,7 @@ class CleanupStats:
     skipped_not_merged: int = 0
     skipped_grace: int = 0
     skipped_uncommitted: int = 0
+    skipped_editable: int = 0
     cleaned_branches: int = 0
     kept_branches: int = 0
     killed_tmux: int = 0
@@ -268,6 +270,94 @@ def cleanup_worktree(
     return True
 
 
+def find_editable_pip_installs(worktree_path: pathlib.Path) -> list[str]:
+    """Find pip packages with editable installs pointing into a worktree.
+
+    Checks all available Python interpreters for packages installed in
+    editable mode (pip install -e) whose source location is inside the
+    given worktree path.
+
+    Args:
+        worktree_path: Absolute path to the worktree directory.
+
+    Returns:
+        List of package names with editable installs inside the worktree.
+    """
+    packages: list[str] = []
+    worktree_str = str(worktree_path.resolve())
+
+    # Find Python interpreters to check
+    interpreters: list[str] = []
+    for name in ("python3", "python"):
+        try:
+            result = subprocess.run(
+                ["which", name],
+                capture_output=True,
+                text=True,
+            )
+            if result.returncode == 0:
+                path = result.stdout.strip()
+                if path and path not in interpreters:
+                    interpreters.append(path)
+        except Exception:
+            continue
+
+    # Also check for venvs inside the worktree itself
+    for venv_dir in worktree_path.glob("*/.venv/bin/python"):
+        venv_python = str(venv_dir)
+        if venv_python not in interpreters:
+            interpreters.append(venv_python)
+    for venv_dir in worktree_path.glob(".venv/bin/python"):
+        venv_python = str(venv_dir)
+        if venv_python not in interpreters:
+            interpreters.append(venv_python)
+
+    for interpreter in interpreters:
+        try:
+            # List all installed packages
+            result = subprocess.run(
+                [interpreter, "-m", "pip", "list", "--editable", "--format=json"],
+                capture_output=True,
+                text=True,
+                timeout=15,
+            )
+            if result.returncode != 0:
+                continue
+
+            try:
+                editable_pkgs = json.loads(result.stdout)
+            except (json.JSONDecodeError, ValueError):
+                continue
+
+            for pkg_info in editable_pkgs:
+                pkg_name = pkg_info.get("name", "")
+                if not pkg_name:
+                    continue
+
+                # Get detailed info to find the editable location
+                show_result = subprocess.run(
+                    [interpreter, "-m", "pip", "show", pkg_name],
+                    capture_output=True,
+                    text=True,
+                    timeout=15,
+                )
+                if show_result.returncode != 0:
+                    continue
+
+                for line in show_result.stdout.splitlines():
+                    if line.startswith("Editable project location:") or line.startswith("Location:"):
+                        location = line.split(":", 1)[1].strip()
+                        if location.startswith(worktree_str):
+                            if pkg_name not in packages:
+                                packages.append(pkg_name)
+                            break
+
+        except (subprocess.TimeoutExpired, FileNotFoundError, OSError):
+            continue
+
+    return packages
+
+
 def clean_worktrees(
     repo_root: pathlib.Path,
     stats: CleanupStats,
@@ -314,6 +404,18 @@ def clean_worktrees(
             if active_pids:
                 log_info(f"  Active process(es) using worktree: {active_pids} - preserving")
                 stats.skipped_in_use += 1
+                continue
+
+        # Check for editable pip installs pointing into the worktree
+        editable_pkgs = find_editable_pip_installs(worktree_path)
+        if editable_pkgs:
+            pkg_list = ", ".join(editable_pkgs)
+            if force:
+                log_warning(f"  Editable pip install(s) found ({pkg_list}) - removing anyway (--force)")
+            else:
+                log_warning(f"  Editable pip install(s) found ({pkg_list}) - skipping")
+                log_info("  Use --force to remove anyway, or uninstall first: pip uninstall " + " ".join(editable_pkgs))
+                stats.skipped_editable += 1
                 continue
 
         # Check issue state via GitHub CLI
@@ -626,6 +728,9 @@ def print_summary(stats: CleanupStats, dry_run: bool = False, safe_mode: bool = 
 
     if stats.skipped_in_use > 0:
         print(f"  Skipped (in use by shepherd): {stats.skipped_in_use}")
+
+    if stats.skipped_editable > 0:
+        print(f"  Skipped (editable pip install): {stats.skipped_editable}")
 
     if safe_mode:
         print(f"  Skipped (open/not merged): {stats.skipped_open + stats.skipped_not_merged}")

--- a/loom-tools/tests/test_clean.py
+++ b/loom-tools/tests/test_clean.py
@@ -2,7 +2,10 @@
 
 from __future__ import annotations
 
+import json
 import pathlib
+import subprocess
+from unittest.mock import patch
 
 import pytest
 
@@ -12,6 +15,7 @@ from loom_tools.clean import (
     _get_dir_size,
     check_grace_period,
     check_uncommitted_changes,
+    find_editable_pip_installs,
     main,
     print_summary,
 )
@@ -108,6 +112,7 @@ class TestCleanupStats:
         stats = CleanupStats()
         assert stats.cleaned_worktrees == 0
         assert stats.skipped_open == 0
+        assert stats.skipped_editable == 0
         assert stats.errors == 0
 
 
@@ -132,6 +137,18 @@ class TestPrintSummary:
         print_summary(stats)
         captured = capsys.readouterr()
         assert "Errors: 2" in captured.out
+
+    def test_print_summary_with_skipped_editable(self, capsys: pytest.CaptureFixture[str]) -> None:
+        stats = CleanupStats(skipped_editable=2)
+        print_summary(stats)
+        captured = capsys.readouterr()
+        assert "Skipped (editable pip install): 2" in captured.out
+
+    def test_print_summary_no_skipped_editable_when_zero(self, capsys: pytest.CaptureFixture[str]) -> None:
+        stats = CleanupStats()
+        print_summary(stats)
+        captured = capsys.readouterr()
+        assert "editable pip install" not in captured.out
 
 
 class TestCLI:
@@ -185,6 +202,235 @@ class TestCLI:
         monkeypatch.chdir(mock_repo)
         result = main(["--dry-run", "--force", "--safe", "--grace-period", "1200"])
         assert result == 0
+
+
+class TestFindEditablePipInstalls:
+    """Tests for find_editable_pip_installs function."""
+
+    def _make_run_side_effect(
+        self,
+        worktree_path: str,
+        editable_packages: list[dict[str, str]],
+        show_outputs: dict[str, str],
+    ):
+        """Create a subprocess.run side effect for mocking pip commands.
+
+        Args:
+            worktree_path: The worktree path to use.
+            editable_packages: List of dicts with "name" and "version" for pip list --editable.
+            show_outputs: Dict mapping package name to pip show stdout.
+        """
+
+        def side_effect(cmd, **kwargs):
+            cmd_str = " ".join(str(c) for c in cmd)
+            result = subprocess.CompletedProcess(cmd, 0)
+
+            if "which" in cmd_str:
+                result.stdout = "/usr/bin/python3\n"
+                result.stderr = ""
+                return result
+
+            if "pip list --editable" in cmd_str:
+                result.stdout = json.dumps(editable_packages)
+                result.stderr = ""
+                return result
+
+            if "pip show" in cmd_str:
+                # Extract package name (last argument)
+                pkg_name = cmd[-1]
+                if pkg_name in show_outputs:
+                    result.stdout = show_outputs[pkg_name]
+                    result.stderr = ""
+                else:
+                    result.returncode = 1
+                    result.stdout = ""
+                    result.stderr = f"WARNING: Package(s) not found: {pkg_name}"
+                return result
+
+            result.returncode = 1
+            result.stdout = ""
+            result.stderr = ""
+            return result
+
+        return side_effect
+
+    def test_editable_install_inside_worktree(self, tmp_path: pathlib.Path) -> None:
+        """Editable install with location inside worktree should be detected."""
+        worktree = tmp_path / "issue-42"
+        worktree.mkdir()
+
+        side_effect = self._make_run_side_effect(
+            str(worktree),
+            editable_packages=[{"name": "loom-tools", "version": "0.1.0"}],
+            show_outputs={
+                "loom-tools": (
+                    f"Name: loom-tools\n"
+                    f"Version: 0.1.0\n"
+                    f"Location: {worktree}/loom-tools/src\n"
+                    f"Editable project location: {worktree}/loom-tools\n"
+                ),
+            },
+        )
+
+        with patch("loom_tools.clean.subprocess.run", side_effect=side_effect):
+            result = find_editable_pip_installs(worktree)
+
+        assert result == ["loom-tools"]
+
+    def test_editable_install_outside_worktree(self, tmp_path: pathlib.Path) -> None:
+        """Editable install with location outside worktree should not be detected."""
+        worktree = tmp_path / "issue-42"
+        worktree.mkdir()
+        other_dir = tmp_path / "other-project"
+        other_dir.mkdir()
+
+        side_effect = self._make_run_side_effect(
+            str(worktree),
+            editable_packages=[{"name": "some-pkg", "version": "1.0.0"}],
+            show_outputs={
+                "some-pkg": (
+                    f"Name: some-pkg\n"
+                    f"Version: 1.0.0\n"
+                    f"Location: {other_dir}/src\n"
+                    f"Editable project location: {other_dir}\n"
+                ),
+            },
+        )
+
+        with patch("loom_tools.clean.subprocess.run", side_effect=side_effect):
+            result = find_editable_pip_installs(worktree)
+
+        assert result == []
+
+    def test_pip_not_found(self, tmp_path: pathlib.Path) -> None:
+        """When pip commands fail, should return empty list without error."""
+        worktree = tmp_path / "issue-42"
+        worktree.mkdir()
+
+        def side_effect(cmd, **kwargs):
+            cmd_str = " ".join(str(c) for c in cmd)
+            if "which" in cmd_str:
+                result = subprocess.CompletedProcess(cmd, 0)
+                result.stdout = "/usr/bin/python3\n"
+                result.stderr = ""
+                return result
+            # All pip commands fail
+            raise FileNotFoundError("pip not found")
+
+        with patch("loom_tools.clean.subprocess.run", side_effect=side_effect):
+            result = find_editable_pip_installs(worktree)
+
+        assert result == []
+
+    def test_pip_returns_error(self, tmp_path: pathlib.Path) -> None:
+        """When pip list returns error code, should return empty list."""
+        worktree = tmp_path / "issue-42"
+        worktree.mkdir()
+
+        def side_effect(cmd, **kwargs):
+            cmd_str = " ".join(str(c) for c in cmd)
+            result = subprocess.CompletedProcess(cmd, 0)
+            result.stderr = ""
+
+            if "which" in cmd_str:
+                result.stdout = "/usr/bin/python3\n"
+                return result
+
+            if "pip list" in cmd_str:
+                result.returncode = 1
+                result.stdout = ""
+                return result
+
+            result.returncode = 1
+            result.stdout = ""
+            return result
+
+        with patch("loom_tools.clean.subprocess.run", side_effect=side_effect):
+            result = find_editable_pip_installs(worktree)
+
+        assert result == []
+
+    def test_no_editable_packages(self, tmp_path: pathlib.Path) -> None:
+        """When no editable packages are installed, should return empty list."""
+        worktree = tmp_path / "issue-42"
+        worktree.mkdir()
+
+        side_effect = self._make_run_side_effect(
+            str(worktree),
+            editable_packages=[],
+            show_outputs={},
+        )
+
+        with patch("loom_tools.clean.subprocess.run", side_effect=side_effect):
+            result = find_editable_pip_installs(worktree)
+
+        assert result == []
+
+    def test_pip_list_returns_invalid_json(self, tmp_path: pathlib.Path) -> None:
+        """When pip list returns invalid JSON, should return empty list."""
+        worktree = tmp_path / "issue-42"
+        worktree.mkdir()
+
+        def side_effect(cmd, **kwargs):
+            cmd_str = " ".join(str(c) for c in cmd)
+            result = subprocess.CompletedProcess(cmd, 0)
+            result.stderr = ""
+
+            if "which" in cmd_str:
+                result.stdout = "/usr/bin/python3\n"
+                return result
+
+            if "pip list" in cmd_str:
+                result.stdout = "not valid json{{"
+                return result
+
+            result.returncode = 1
+            result.stdout = ""
+            return result
+
+        with patch("loom_tools.clean.subprocess.run", side_effect=side_effect):
+            result = find_editable_pip_installs(worktree)
+
+        assert result == []
+
+    def test_location_field_without_editable_prefix(self, tmp_path: pathlib.Path) -> None:
+        """Detect via 'Location:' field when 'Editable project location:' is absent."""
+        worktree = tmp_path / "issue-42"
+        worktree.mkdir()
+
+        side_effect = self._make_run_side_effect(
+            str(worktree),
+            editable_packages=[{"name": "my-pkg", "version": "0.1.0"}],
+            show_outputs={
+                "my-pkg": (
+                    f"Name: my-pkg\n"
+                    f"Version: 0.1.0\n"
+                    f"Location: {worktree}/my-pkg/src\n"
+                ),
+            },
+        )
+
+        with patch("loom_tools.clean.subprocess.run", side_effect=side_effect):
+            result = find_editable_pip_installs(worktree)
+
+        assert result == ["my-pkg"]
+
+    def test_no_python_interpreters(self, tmp_path: pathlib.Path) -> None:
+        """When no Python interpreters found, should return empty list."""
+        worktree = tmp_path / "issue-42"
+        worktree.mkdir()
+
+        def side_effect(cmd, **kwargs):
+            # which fails for all interpreters
+            result = subprocess.CompletedProcess(cmd, 1)
+            result.stdout = ""
+            result.stderr = ""
+            return result
+
+        with patch("loom_tools.clean.subprocess.run", side_effect=side_effect):
+            result = find_editable_pip_installs(worktree)
+
+        assert result == []
 
 
 class TestDocumentedDivergences:

